menas/src/main/scala/za/co/absa/enceladus/menas/services/SchemaRegistryService.scala

```diff
@@ -61,6 +64,48 @@ class SchemaRegistryService @Autowired()() {
     }
   }
 
+  /**
+   * JKS based TrustManagerFactory option based on [[SecureConfig.Keys.javaxNetSslTrustStore]] and
+   * [[SecureConfig.Keys.javaxNetSslTrustStorePassword]] presence in [[config]]
+   */
+  lazy val trustManagerFactory: Option[TrustManagerFactory] = {
+    if (SecureConfig.hasTrustStoreProperties(config)) {
+      val trustStore = KeyStore.getInstance(defaultStoreType)
+
+      val tsInputStream = new FileInputStream(config.getString(SecureConfig.Keys.javaxNetSslTrustStore))
```

Comment:
Is this safe? What if it does not exist?

Reply:
`SecureConfig.hasTrustStoreProperties(config)` should guard you for this very case.

Reply:
Sorry, I mean what is the load fails somehow. Does it leave an open stream?

Reply:
Keen observation, I have put the closing in `finally`.